x/payment/types/square_sizes_test.go

```diff
@@ -0,0 +1,44 @@
+package types
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestAllSquareSizes(t *testing.T) {
+	type testCase struct {
+		msgSize int
+		want    []uint64
+	}
+
+	tests := []testCase{
+		{0, []uint64{2, 4, 8, 16, 32, 64, 128}}, // should this function return an error for a message size of 0?
```

Comment:
this is a really good point! I'm not sure. Technically, iirc, messages of length zero are invalid, but handling that error each time we use this function seems cumbersome. Perhaps we should just handle that in validate basic (if we're not already) and then assume that its fine here.

Reply:
I pulled the trigger on merge before seeing this but agreed seems cumbersome to handle errors from this function and would prefer a validate basic check on message length.